.github/workflows/pdk-basic.yml

```diff
@@ -0,0 +1,51 @@
+# This is a basic workflow to help you get started with Actions
+
+name: main
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push 
+  push:
+  workflow_dispatch:
+
+env:
+  GITHUB_USER: ${{ secrets.PUPPET_REPO_USER }}
+  GITHUB_TOKEN: ${{ secrets.PUPPET_REPO_USER_TOKEN }}
+  
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  validate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    container: puppet/puppet-dev-tools:2022-03-28-92c7176
```

Comment:
You can make this an input so it's only defined once and then callers can override it easily.

Reply:
Awesome.  Only thing I can't do is allow user to pass in matrix.    Not sure if people want to support 5,6,7 and also need to control when pdk stops supporting 5.